glusterd2/commands/peers/addpeer.go

```diff
@@ -141,6 +141,7 @@ func createPeerAddResp(p *peer.Peer) *api.PeerAddResp {
 	return &api.PeerAddResp{
 		ID:              p.ID,
 		Name:            p.Name,
+		Online:          true,
```

Comment:
I think instead of hardcoding online as true, can we check liveness status of peer?

Reply:
@oshankkumar This is during the peer add, I don't think we need to check the liveness status.
if the node is available then only we can make that peer add is successful.

do we need to check liveness status during peer add also?

Reply:
@Madhu-1  If we are considering that there is no possibility that node will go down between the time we call joinCluster() and returning the response of peer add, then hard coding true is fine, even when large number of peers are added parallely.

Reply:
I agree with @Madhu-1, if Peer add successful then that means Peer is online. Peer can go down after peer add, but that status can be obtained from Peer status.

Reply:
if glusterd2 goes down in between it will take 5 seconds to make the peer state as offline.
even if you check the liveness check it won't make much difference, liveness timeout is around 5 seconds.

even if you get so many requests at a time, gd2 won't take 5 seconds to respond back.

for the corner cases, I have added the check, but it won't make much difference, for me, it will take some more time to send back the response as it includes the get request from ETCD

Reply:
let me know it make sense to add the check

Reply:
Not required I think, If we say peer add is successful and online as false then again it adds confusion as now.